client/blocks/utils.js

```diff
@@ -20,14 +20,16 @@ export const getBlocksConfiguration = () => {
  * Determines if off-session payment should be set up.
  *
  * @param {boolean} showSaveOption - Whether to show the save option.
+ * @param {boolean} paymentMethodReusable - Whether the payment method is reusable.
  * @return {boolean} True if off-session payment should be set up, false otherwise.
  */
-export const shouldSetupOffSessionPayment = ( showSaveOption ) => {
-	const config = getBlocksConfiguration();
-	const hasAutoRenewingSubscription =
-		config?.cartContainsSubscription &&
-		! config?.subscriptionRequiresManualRenewal;
-	return hasAutoRenewingSubscription || showSaveOption;
+export const shouldSetupOffSessionPayment = (
+	showSaveOption,
+	paymentMethodReusable
+) => {
+	return (
+		hasAutoRenewingSubscription( paymentMethodReusable ) || showSaveOption
```

Comment:
Another small, non-blocking nit: I would also prefer the variable naming to make it super-clear that the values are booleans.
```suggestion
export const shouldSetupOffSessionPayment = (
	shouldShowSaveOption,
	isPaymentMethodReusable
) => {
	return (
		hasAutoRenewingSubscription( isPaymentMethodReusable ) || shouldShowSaveOption
```

Reply:
I could not auto-apply your suggestions, but I manually did them in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4401/commits/e53a28ee1cbc17fc0b661c8e74a45c24c31564cb